server/src/main/java/com/paypal/selion/grid/servlets/transfer/DefaultManagedArtifact.java

```diff
@@ -48,29 +47,95 @@
 
     private static final String HTTP_CONTENT_TYPE = "application/zip";
 
-    private String filePath = null;
-
     private File artifactFile = null;
 
     private String artifactName = null;
 
-    private String folderName = null;
+    private String subFolderName = null;
 
-    private String parentFolderName = null;
+    private String uuidFolderName = null;
 
     private byte[] contents = null;
 
-    private final long timeToLiveInMillis;
+    private static final long timeToLiveInMillis = ConfigParser.parse().getLong(EXPIRY_CONFIG_PROPERTY);
+
+    private String requestPathInfo;
+
+    private DefaultCriteria requestedCriteria = null;
+
+    final class DefaultRequestParameters implements RequestParameters {
+        public static final String USER_ID = "userId";
 
-    public DefaultManagedArtifact(String pathName) {
-        this.filePath = pathName;
-        artifactFile = new File(this.filePath);
-        timeToLiveInMillis = ConfigParser.parse().getLong(EXPIRY_CONFIG_PROPERTY);
+        public Map<String, Boolean> getParameters() {
+            Map<String, Boolean> headers = new HashMap<String, Boolean>();
+            headers.put(ARTIFACT_FILE_NAME, true);
+            headers.put(ARTIFACT_FOLDER_NAME, false);
+            headers.put(USER_ID, true);
+            return headers;
+        }
+
+        public boolean isRequired(String parameter) {
+            if (getParameters().containsKey(parameter)) {
+                return getParameters().get(parameter).booleanValue();
+            }
+            return false;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T extends RequestParameters> T getRequstParameters() {
+        return (T) new DefaultRequestParameters();
+    }
+
+    public DefaultManagedArtifact() {
         if (LOGGER.isLoggable(Level.FINE)) {
-            LOGGER.log(Level.FINE, "Time To Live configured in Grid: " + timeToLiveInMillis + " milli seconds.");
+            LOGGER.log(Level.FINE, "Managed Artifact TTL configured in Grid to " + timeToLiveInMillis + " milliseconds.");
+        }
+    }
+
+    // package visible, this constructor is for test purposes only
+    DefaultManagedArtifact(String pathName) {
+        this();
+        initFromPath(pathName);
+    }
+
+    public void initFromPath(String pathName) {
+        // do not allow path that are empty or try to break out of the repository folder
+        Preconditions.checkArgument(StringUtils.isNotBlank(pathName));
+        Preconditions.checkArgument(!StringUtils.contains(pathName, ".."));
+
+        String repoPath = ManagedArtifactRepository.getInstance().getRepositoryFolder().getAbsolutePath();
+        String filePath = pathName;
+        if (!pathName.contains(repoPath)) {
```

Comment:
This check can fail on windows where the `repoPath` is on a different drive that the drive included in `pathName`.. Will fix this.